### DISCUSSION TITLE
Check if model is already loaded

### DISCUSSION THREAD

**Discussion by GavasHaluk** [openai/whisper#1306] [Q&A] [answered]
Is there a way to check if a model is already loaded and skip the `load_model` step? I have seen that the load_model is invoked multiple times between my endpoint calls which is not ideal since it adds quite a bit of execution time especially with using larger models.

**Comment by baibhavKumar1** [comment]
Yes just add a flag , when you add it for first time, and if it still
exists don't load the model again. Check if this works, I'm not sure.

**Comment by jongwook** [comment, marked as answer]
Something like [`functools.cache`](https://docs.python.org/3/library/functools.html#functools.cache) can be handy for this purpose.

**Comment by xavierskip** [comment]
how to use [functools.cache](https://docs.python.org/3/library/functools.html#functools.cache)  in this case?
I try
``` python
import whisper
from functools import cache

@cache
def load_model(model):
    return whisper.load_model(model)

whisper_model = load_model(Model)
```
seems not work.
